### PR TITLE
[PY] feat: Add managed identity auth support to `AssistantsPlanner`

### PR DESCRIPTION
**Pull request by singhk97**
## Linked issues

closes: #minor
tracking: #1918

## Details
Add ability to authenticate `AssistantsPlanner` using Azure Managed Identity by providing a `azure_ad_token_bearer` field in the initializer options.

#### Change details
Updated samples
- math bot
- order bot

## Attestation Checklist

- [x] My code follows the style guidelines of this project

- I have checked for/fixed spelling, linting, and other errors
- I have commented my code for clarity
- I have made corresponding changes to the documentation (updating the doc strings in the code is sufficient)
- My changes generate no new warnings
- I have added tests that validates my changes, and provides sufficient test coverage. I have tested with:
  - Local testing
  - E2E testing in Teams
- New and existing unit tests pass locally with my changes

### CODE REVIEW

File: python/packages/ai/tests/ai/planners/test_assistants_planner.py [hunk 1]
```diff
@@ -573,6 +578,7 @@ async def test_create_azure_openai_assistant(self, mock_async_azure_openai):
 
         assistant = await AssistantsPlanner.create_assistant(
             api_key="",
+            azure_ad_token_provider=None,
```
Comment (lilyydu) [inline python/packages/ai/tests/ai/planners/test_assistants_planner.py:581]: could we add 1-2 tests to check this flow, when its not 'None'?

Comment (singhk97) [reply python/packages/ai/tests/ai/planners/test_assistants_planner.py:581]: done

**Review by lilyydu — APPROVED** [review]
